### PR TITLE
ghc 9.4

### PR DESCRIPTION
**Pull request by andreasabel**
- Compatibility with aeson-2.0
- Compatiblity with Cabal-3.4

Tested build with these LTSs:
- 18.28 (current)
- 19.33
- 20.26
- 21.0 (new)

### CODE REVIEW

File: src/Curator/Snapshot.hs [hunk 1]
```diff
@@ -412,14 +438,22 @@ getPkgInfo constraints compilerVer pname sp = do
       { piVersion = snapshotVersion (spLocation sp)
       , piAllDeps = allDeps
       , piTreeDeps = treeDeps
+#if MIN_VERSION_Cabal(3,4,0)
+      , piCabalVersion = C.specVersion $ C.packageDescription gpd
+#else
       , piCabalVersion = C.specVersion $ C.packageDescription gpd
+#endif
```
Comment (andreasabel) [inline src/Curator/Snapshot.hs:445]: Oops, this should go.

File: src/Curator/Types.hs [hunk 2]
```diff
@@ -21,6 +23,20 @@ import qualified RIO.Map as Map
...
+#else
+type Key = Text
+#endif
+
+toKey :: Text -> Key
+#if MIN_VERSION_aeson(2,0,0)
+toKey = Key.fromText
+#else
+toKey = id
+#endif
```
Comment (andreasabel) [inline src/Curator/Types.hs:38]: `toKey` seems unnecessary.

File: src/Curator/Types.hs [hunk 3]
```diff
@@ -21,6 +23,13 @@ import qualified RIO.Map as Map
 import qualified RIO.Set as Set
 import RIO.Time
 
+#if MIN_VERSION_aeson(2,0,0)
+import Data.Aeson.Key (Key)
+import qualified Data.Aeson.Key as Key
```
Comment (andreasabel) [inline src/Curator/Types.hs:28]: import not needed

**Comment by andreasabel** [comment]
I tested it according to the recipe in https://github.com/commercialhaskell/stackage/pull/7029#issuecomment-1609132504 and `curator` worked correctly AFAICT.

PR is ready @snoyberg .

**Review by snoyberg — APPROVED** [review]
Thanks!

**Pull request merged**